### ISSUE TITLE
The Speaker Details page URL should contain the speaker's username rather than numerical ID

### ISSUE THREAD

**Issue by alastairs** [dddeastanglia/DDDEastAnglia#84]
The current speaker details/profile page URL is not terribly friendly, e.g., http://www.dddeastanglia.com/Speakers/Details/23. It would be better to use the username rather than a numerical id as the identifying part of the URL; it might also be nice to rename the Details action to Profile.

**Comment by alastairs** [comment]
I made an attempt at this, but it looks like the T4MVC stuff is actively blocking the renaming of the Details action to Profile, because there's a Profile controller. This is madness.

**Comment by ninjaferret** [comment]
I'd like to get rid of the T4MVC stuff too.

It would definitely be nice for SEO (if we really need it) and we could do the same using session titles perhaps format the URL similar to stackoverflow? (looks like id/title where the title has had non-word characters replaced with dashes)

**Comment by adrianbanks** [comment]
Yes, SO appends the title to the end of the url for SEO, but it has no real use other than that (take it off and the url still works).

**Comment by alastairs** [comment]
What if we got rid of the ID altogether? Any objections? The SO **profile** URLs are a bit annoying because they look like they're easy to guess, but they're not. I'm thinking we should aim for `/Speakers/Profile/alastairs` for profile URLs. 

I'm ok with including the ID in the session URL because there's a small chance two people might submit sessions with the same title, particularly once its been through the necessary munging process.

**Comment by adrianbanks** [comment]
Sounds reasonable. However, people will only need to link to their profile when touting their sessions, which will happened only up until the end of voting. We don't want to change the urls after voting starts (and ideally we should have them pinned down by now), so if this isn't going to get done by then, I suggest that we leave it altogether until after the event.

**Comment by alastairs** [comment]
We could redirect from the existing URLs? Might not be worth the hassle of setting up. I think I can get this change done pretty quickly once #107 is merged.

**Comment by adrianbanks** [comment]
We don't want to remove the existing urls as people are already using them to link to their profile page. Adding another alias is not a problem, but I would still question the value of doing it now.
